packages/trader/src/App/Components/Elements/__tests__/market-countdown-timer.spec.tsx

```diff
@@ -75,21 +73,24 @@ describe('<MarketCountdownTimer />', () => {
     };
 
     it('should not render component with children if is_main_page is true', () => {
-        const new_props = { ...mock_default_props };
-        new_props.is_main_page = true;
+        const new_props = { ...mock_default_props, is_main_page: true };
         const { container } = render(mockMarketCountdownTimer(mockStore(default_mock_store), new_props));
 
         expect(container).toBeEmptyDOMElement();
     });
     it('should render component with children if is_main_page is false', async () => {
         jest.useFakeTimers();
+        const { rerender } = render(mockMarketCountdownTimer(mockStore(default_mock_store), mock_default_props));
         await waitFor(() => {
-            const { rerender } = render(mockMarketCountdownTimer(mockStore(default_mock_store), mock_default_props));
             rerender(mockMarketCountdownTimer(mockStore(default_mock_store), mock_default_props));
         });
-        jest.runOnlyPendingTimers();
-
-        expect(screen.getByText('It will reopen at')).toBeInTheDocument();
-        expect(screen.getByText('Please come back in')).toBeInTheDocument();
+        await act(async () => {
+            jest.runOnlyPendingTimers();
+        });
+        await waitFor(() => {
+            expect(screen.getByText('It will reopen at')).toBeInTheDocument();
+            expect(screen.getByText('Please come back in')).toBeInTheDocument();
+        });
+        jest.useRealTimers();
```

Comment:
Also, I've improved these tests but I didn't add a check for <br  because it has no role and jest complains about direct access to DOM when I try.

Reply:
Thank you for improving the tests as timers were always challenging for me to test)))